orc/src/main/java/org/apache/iceberg/data/orc/GenericOrcWriters.java

```diff
@@ -531,6 +543,32 @@ public void nonNullWrite(int rowId, Map<K, V> map, ColumnVector output) {
     }
   }
 
+  private static class PositionDeleteWriter implements OrcRowWriter<PositionDelete<Record>> {
+    private final OrcRowWriter<Record> rowWriter;
+    private final GenericRecord record;
+
+    PositionDeleteWriter(Schema deleteSchema, OrcRowWriter rowWriter) {
+      this.rowWriter = rowWriter != null ?
+          rowWriter : GenericOrcWriter.buildWriter(deleteSchema, ORCSchemaUtil.convert(deleteSchema));
+      this.record = GenericRecord.create(deleteSchema);
+    }
+
+    @Override
+    public void write(PositionDelete<Record> row, VectorizedRowBatch output) throws IOException {
+      record.set(0, row.path());
```

Comment:
Nit / non-blocking: Do we have a named constant for the 0 / 1 record position (for `path` and `pos` and I guess `row` for 2)?

If we have that elsewhere, would be nice to see it used.

Reply:
We use this in Parquet and ORC, but my feeling that this should be limited to the specific FileFormats